src/prefect/tasks/prometheus/pushgateway.py

```diff
@@ -0,0 +1,137 @@
+from enum import Enum
+from prometheus_client import (
+    CollectorRegistry,
+    Gauge,
+    pushadd_to_gateway,
+    push_to_gateway,
+)
+from typing import Dict, List, Optional
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+class PushMode(Enum):
+    Push = 1
+    PushAdd = 2
+
+
+class PushGaugeToGateway(Task):
+    """
+    Task that allow you to push a Gauge to prometheus [PushGateway]
+    (ttps://prometheus.io/docs/practices/pushing/).
+
+    Args:
+        - pushgateway_url (str, optional): Url of the prometheus pushgateway instance
+        - counter_name (str, optional): Name of the counter
+        - counter_description (str, optional): description of the counter
+        - grouping_key (str, optional): List of the key used to calculate the grouping key
+        - job_name (str, optional): Name of the job
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+
+    """
+
+    def __init__(
+        self,
+        pushgateway_url: str = None,
+        counter_name: str = None,
+        counter_description: str = None,
+        grouping_key: List[str] = None,
+        job_name: str = None,
+        **kwargs,
+    ):
+        self.pushgateway_url = pushgateway_url
+        self.counter_name = counter_name
+        self.counter_description = counter_description
+        self.job_name = job_name
+        self.grouping_key = grouping_key
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs(
+        "pushgateway_url",
+        "counter_name",
+        "counter_description",
+        "job_name",
+        "grouping_key",
+    )
+    def run(
+        self,
+        values: List[float] = [],
+        labels: List[Dict[str, str]] = [],
+        action: PushMode = PushMode.Push,
```

Comment:
does it make sense to have the option to provide `action` at `init` or `run` time too?

Reply:
it could true. I think having two tasks may be better as well. What do you think?

Reply:
I think using the default `action` from `init` would make the most sense